sdk/src/main/java/com/okta/spring/boot/sdk/OktaSdkConfig.java

```diff
@@ -108,19 +111,40 @@ protected ClientCredentials oktaSdkClientCredentials() {
     }
 
     /**
-     * Spring Boot conditional based on the existance of the {code}okta.client.token{code} property.
+     * Spring Boot conditional based on the existence of the properties:
+     * <pre>{@code
+     *  - okta.client.token
+     *  - okta.client.orgUrl
+     * }</pre>
+     *
+     * <p> If <code>'okta.client.orgUrl'</code> is absent, try to resolve it from <code>'okta.oauth2.issuer'</code>.
      */
-    static class OktaApiTokenCondition extends SpringBootCondition {
+    static class OktaApiConditions extends SpringBootCondition {
+
+        private static final Pattern ORG_URL_PATTERN = Pattern.compile("^(?<orgUrl>http(s)?://[^/]+)/?.*$");
```

Comment:
Use `ConfigurationValidator.validateOrgUrl` here instead of using a custom regex.

https://github.com/okta/okta-commons-java/blob/master/config-check/src/main/java/com/okta/commons/configcheck/ConfigurationValidator.java

Reply:
Thank you. I saw this validator. But my aim is not to validate the url here. I used regex only to grab `orgUrl` part.
If you want, we can add additional validation before `matcher.find()`, but not instead. Would you like?

Reply:
..or I'll use `String orgUrl = new URL(new URL(issuer), "/").toString();` instead

Reply:
Ahh, I see, I think using the URL/URI methods would be more clear.